tests/unit/useViolationsTest.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable rulesdir/prefer-underscore-method -- Disabled because it incorrectly flags the use of `it.each` */
+import { renderHook } from '@testing-library/react-native';
+import { useViolations, violationFields } from '@libs/Violations/useViolations';
+
+
+// Derive test cases from the violationFields object
+const allViolations = Object.keys(violationFields).map((name) => ({name}));
+// A list of all the field names that appear in `violationFields`
+const fieldNames = [...new Set(Object.values(violationFields))];
+// create an index of violations by field name
+const violationsByField = Object.entries(violationFields).reduce(
+    (acc, [violation, field]) => ({
+        ...acc,
+        [field]: [...(acc[field] || []), {name: violation}],
+    }),
+    {},
+);
```

Comment:
This is very hard to read, and the one above isn't great. Can we iterate over `Object.entries(violationFields)` and set allViolations, and violationsField like that?

Also, can fieldNames be set using the `ViolationField` type defined in useViolations?

Reply:
> can fieldNames be set using the ViolationField type defined in useViolations?

No types aren't accessible at runtime in typescript (and this is a javascript file).

I'll simplify the rest.

Reply:
Should be more readable in the latest version.